lib/api/model/model.dart

```diff
@@ -596,7 +596,10 @@ extension type const TopicName(String _value) {
 /// Different from [MessageDestination], this information comes from
 /// [getMessages] or [getEvents], identifying the conversation that contains a
 /// message.
-sealed class Conversation {}
+sealed class Conversation {
+  /// Whether [this] and [other] have the same canonical form.
+  bool isSameAs(Conversation other);
```

Comment:
```suggestion
  /// Whether [this] and [other] refer to the same Zulip conversation.
  bool isSameAs(Conversation other);
```

I'd think of a "canonical form" as a strategy for implementing this sort of method, and the intended semantics as something a bit more abstract.